core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -90,4 +95,58 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeGenericValue(obj);
         }
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startArray(name);
+        for (Object value : values) {
+            builder.value(value);
+        }
+        builder.endArray();
+        return builder;
+    }
+
+    public static GetField fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() != XContentParser.Token.FIELD_NAME) {
+            throw new ParsingException(parser.getTokenLocation(),
+                    "expected " + XContentParser.Token.FIELD_NAME + " - found " + parser.currentToken());
+        }
+        String fieldName = parser.currentName();
+        List<Object> values = new ArrayList<>();
+        XContentParser.Token token = parser.nextToken();
+        if (token != XContentParser.Token.START_ARRAY) {
+            throw new ParsingException(parser.getTokenLocation(),
+                    "expected " + XContentParser.Token.START_ARRAY + " - found " + parser.currentToken());
+        }
+        while(parser.nextToken() != XContentParser.Token.END_ARRAY) {
+            values.add(parser.objectText());
```

Comment:
good point. I have tested geo points manually and we don't actually print anything out if any or those are requested via stored_fields, which makes me think that they are not supported in GetField, but I had exactly the same doubts. Printing things out is always easy as we have all the info (class name etc.) to decide what to do, when reading back it is much trickier as we only have the json data type information. We do end up not being able to parse back what we write. It may be better to change how we print things out... like simply printing out objects by calling their toString, but I am not sure if that's a good idea at the moment